sdks/nodejs/src/agonesSDK.js

```diff
@@ -0,0 +1,101 @@
+const grpc = require('grpc');
+
+const messages = require('../lib/sdk_pb');
+const services = require('../lib/sdk_grpc_pb');
+
+class AgonesSDK {
+	constructor() {
+		this.client = new services.SDKClient('localhost:50051', grpc.credentials.createInsecure());
+	}
+
+	async ready() {
+		let request = new messages.Empty();
+		return new Promise((resolve, reject) => {
+			this.client.ready(request, (error, response) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(response.toObject());
+				}
+			});
+		});
+	}
+
+	async shutdown() {
+		let request = new messages.Empty();
+		return new Promise((resolve, reject) => {
+			this.client.shutdown(request, (error, response) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(response.toObject());
+				}
+			});
+		});
+	}
+
+	health() {
+		if (!this.healthStream) {
+			this.healthStream = this.client.health((error) => {
+				if (error) {
```

Comment:
It wasn't clear to me whether this callback ever includes a response, and if it did how to handle that as the callback may be executed some time after the initial call to `agonesSDK.health()`. It's also unclear where this error could be caught if it did not happen in the call chain of the first call to `agonesSDK.health()`.
Another option I considered is to move the stream setup to the constructor.

Reply:
Execution is single threaded so this method should be OK. I'm just concerned that although `this.client.health()` returns immediately, it might hang on to the callback and call with an error later. I can take another look and potentially remove

Reply:
I guess if that happens, then everything breaks - and well, the server becomes unhealthy - which is probably correct anyway?

Reply:
I did a test and removed the throw as this could never be caught by client code.